modelci/app/v1/endpoints/model.py

```diff
@@ -33,3 +33,31 @@ def get_all_model(name: str = None, framework: Framework_ = None, engine: Engine
 def get_model(*, id: str):  # noqa
     model = ModelService.get_model_by_id(id)
     return ModelDetailOut.from_bo(model)
+
+
+@router.get('/structure/{id}')  # TODO: add response_model
+async def get_model_structure(id: str):  # noqa
+    """
+    Get model structure as a DAG.
```

Comment:
as a DAG?

I understand that your target is to obtain the model structure with layer information. Using DAG may be a too strong claim for this scenario.

Reply:
Should be narrow down to a graph with nodes (layers) and edges (connections between layers)